src/compiling/lexing/reader.rs

```diff
@@ -1,11 +1,15 @@
-#![allow(dead_code)]
+#[cfg(feature = "serde")]
+use serde::{Serialize, Deserialize};
+
 const BEGINNING: (usize, usize) = (0, 1);
 
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub enum ReadMode {
     History,
     Future
 }
 
+#[cfg_attr(feature = "serde", derive(Serialize))]
```

Comment:
Should this be `derive(Serialize, Deserialize)` like all the other `cfg_attr` annotations?